acstools/tests/test_wfc_destripe.py

```diff
@@ -26,7 +26,7 @@ def _destripe_one(self, inputfile, outsuffix, outputfile, truthfile,
                                mask1=masks[0], mask2=masks[1])
 
         # Compare results
-        self.compare_outputs([(outputfile, truthfile)], rtol=1e-6)
+        self.compare_outputs([(outputfile, truthfile)], rtol=1e-6, atol=0.1)
```

Comment:
@jryon , is setting `atol=0.1` controversial? Anything lower would cause some comparison to fail. Do you care about disagreement smaller than this in the pixel values?

Reply:
So including the atol causes these tests not to fail anyore? My issue with an absolute tolerance is that the SCI arrays in the files being tested cover a lot of dynamic range, from what I could see. Some have pixel values of order 0.05 electrons and others are 100 electrons. Why not increase rtol or redefine truth instead?

Reply:
If we only want to touch rtol, it would have to be something like `rtol=1e-3` or larger. Here is one of the disagreements:

```
                    a> np.float32(21.261755)
E                    ?                ^ ---
E                   b> np.float32(21.25891)
E                    ?                ^^^
```

This disagreement only shows up using unreleased versions of upstream dependencies (probably because numpy 2.0.dev changed a bunch of stuff, but also could be scipy or astropy here, so not 100% sure). So if you update the truth, the other jobs using released versions would fail instead.

Another option is for someone on the ACS Team to spend time investigating why the results are different using unreleased numpy/scipy/astropy etc.

Reply:
I don't see that disagreement in the previous error log you shared, and I don't know where to find the others - can you point me to them? This one is slightly larger than discrepancies I saw, so I'd like to take a look at all of the errors.

Forgive my naivety, but why are we testing against unreleased versions as well as released versions of dependencies? Why not just test against released versions and update truth/fix problems at that point?

Reply:
> I don't see that disagreement in the previous error log you shared

The way fitsdiff works, you won't see everything. It only popped up for me when I started relaxing the tolerance bit by bit in a local run after I managed to set up the environment locally so that the tests fail in the same way as CI.

> why are we testing against unreleased versions as well as released versions of dependencies?

It is to future-proof the package. Both numpy 2.0 and astropy 6.0 (both major releases but numpy 2.0 is the major refactor one) are slated to be released by the end of the year (so in a month or two). To have a matrix of both released and unreleased versions of dependencies would ensure (1) your code is compatible with things we know are getting released in the future, and (2) any changes to fix 1 won't accidentally break backward compatiblity with already released versions. This is more or less a standard practice in larger libraries that I try to emulate here.

> Why not just test against released versions and update truth/fix problems at that point?

That is also an option you can consider. However, this means you need to bump the minimum required versions to the latest versions almost immediately after they are released. Given how breaking numpy 2.0 is going to be, not all your other dependencies can catch up quickly, so you might be stuck with a very broken CI for many months.

Can you also update the truth without bumping required minimum versions? Yes, you can, but it is a promise you cannot keep because you no longer is able to test against the minimum versions you are advertising due to truth updates.

Does this make sense?

Reply:
@pllim Yes, this makes sense. Thanks for explaining. I certainly don't want to change what we're doing if it's essentially standard practice, and it sounds like a solid system. I'm happy to go with an rtol bump instead of atol, and I'll discuss again with Norman to make sure we're still ok with discrepancies slightly larger than the logs say.